pkg/server/server_credential_test.go

```diff
@@ -31,6 +31,109 @@ func TestCredentialAPI(t *testing.T) {
 
 	for _, test := range testutil.TestDatabases {
 		t.Run(test.Name, func(tt *testing.T) {
+			tt.Run("Batch Update Credential Status", func(ttt *testing.T) {
```

Comment:
we need a few different test cases here

1. batch of 0
2. batch of 1-n (all valid)
3. batch of 1-n (mixed invalid and valid credentials, invalid = cred doesn't exist, cred status already set)

Another test we might need here and elsewhere is performing an update where the key no longer exists (revoked/deleted)

Reply:
Added tests for all batch behaviors.

> Another test we might need here and elsewhere is performing an update where the key no longer exists (revoked/deleted)

I'm unsure about some of the details of the behavior you're trying to test. Mind creating an issue and we can discuss?